htlcswitch/link.go

```diff
@@ -2293,6 +2297,36 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			}
 		}
 
+		// If we need to send out an Stfu, this would be the time to do
+		// so.
+		localOnRemote := l.channel.NumLocalUpdatesPendingOnRemote()
```

Comment:
See prior commit re just using `NeedsCommit` here to decide if we can send or not.

Reply:
Let's collapse this discussion to [this thread](https://github.com/lightningnetwork/lnd/pull/8270#discussion_r1426012653).